Public/Src/Tools/Execution.Analyzer/Analyzers.Core/XLGPlusPlus/Events.proto

```diff
@@ -22,31 +34,845 @@ message EventTypeQuery{
     string UUID = 99;   
 }
 
-// The BXL Invocation Event message
-message BXLInvocationEvent{
+//
+// Enums and Helper Structs
+//
 
-    // Unique identifier for the event - required, TODO: Remove as it may not be needed.
-    string UUID = 1;
+// Note that enum values use C++ scoping rules, meaning that enum values are siblings of their type, 
+// not children of it. Therefore, enum values must be unique within "BuildXL.Execution.Analyzer", not 
+// just within a particular enum. If there are conflicting enum values, append the name of the enum in front.
+// ie: ChooseWorkerCpu ---> PipExecutionStep_ChooseWorkerCpu
+
+enum ExecutionEventId_XLGpp{

```

Comment:
>_XLGpp [](start = 21, length = 6)

Any particular reason for `_XLGpp` for suffix?

Reply:
This was to differentiate the auto-generated class names from the BXL classes of the same name, which caused some conflicts where we would have otherwise had to specify the entire namespace of the class to reference the correct one. (XLGpp stands for XLG++)

Reply:
I start to dislike this suffix.... we need to think of a better naming convention.... what about changing default namespace to something like 'BuildXL.Execution.Analyzer.NEW_NAMESPACE', so we would not need the suffix at all?

---
In reply to: [308462155](https://github.com/microsoft/BuildXL/pull/671#discussion_r308462155) [](ancestors = 308462155)